torch/ao/sparsity/experimental/data_sparsifier/base_data_sparsifier.py

```diff
@@ -0,0 +1,76 @@
+from typing import Optional, Tuple, List, Any, Dict
+from ...sparsifier import base_sparsifier
+from collections import defaultdict
+from torch import nn
+
+
+class Container(nn.Module):
+    def __init__(self):
+        super().__init__()
+
+
+class BaseDataSparsifier(base_sparsifier.BaseSparsifier):
+    r"""
+    Base Data Sparsifier class for all Data sparsifiers.
+    The abstract class accepts raw torch tensors / embedding / embedding bags (refer to SUPPORTED_TYPES above)
+    to prepare for sparsification.
+    In this case, mask (and parametrizations) is owned by the class and not by the user.
+    Specifically, the container object inside the class maintains the mask and parametrizations of the input data
+    Args:
+        - data_list [List[Tuple[str, Any]]]: list of (name, data) tuples to sparsify.
+        Currently - operates on embeddings and torch tensors, parameters
+        Internally, a container module handles the data sparsification.
+
+        - defaults [dict]: default configurations will be attached to the
+            configuration. Only the keys that don't exist in the `config` will
+            be updated.
```

Comment:
1. Don't add `-` in front of the parameter name
2. The data type should be in regular parenthesis `(` and `)`
3. The data type should be human readable, not necessarily `typing` instance. For example, use `list of int's` instead of `List[int]`
4. The continuation in the line should be indented.

One option to rewrite this:

```
    Args:
        data_list (list of tuples):
            list of (name, data) tuples to sparsify.
            Currently - operates on embeddings and torch tensors, parameters
            Internally, a container module handles the data sparsification.
        defaults [dict]:
            default configurations will be attached to the
            configuration. Only the keys that don't exist in the `config` will
            be updated.
```

Reply:
Noted :)

Reply:
modified